tests/tracking/test_tracking.py

```diff
@@ -398,9 +398,32 @@ def test_log_artifact_with_dirs(tracking_uri_mock):
         artifact_uri = mlflow.get_artifact_uri()
         run_artifact_dir = local_file_uri_to_path(artifact_uri)
         mlflow.log_artifact(art_dir)
-        print(os.listdir(run_artifact_dir))
         assert os.listdir(run_artifact_dir) == [os.path.basename(art_dir)]
 
+    # Test log artifact with directory and specified parent folder
+    art_dir = tempfile.mkdtemp()
```

Comment:
You can actually use pytest's `tmpdir` fixture to make these temporary test directories ([see pytest docs](http://doc.pytest.org/en/latest/tmpdir.html#the-tmpdir-fixture)), which will automatically clean them up after the test runs. The `tmpdir` fixture also makes it easy to write files under the temporary directory - it'd be good to write a simple file under the temporary directory in these tests & verify that it's also logged with the right contents.